graalvm/client-hotrod/src/test/java/org/infinispan/client/hotrod/graalvm/substitutions/SubstituteClusteredIT.java

```diff
@@ -0,0 +1,19 @@
+package org.infinispan.client.hotrod.graalvm.substitutions;
+
+import org.infinispan.server.functional.ClusteredIT;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+public class SubstituteClusteredIT {
+}
+
+@TargetClass(ClusteredIT.class)
+final class Target_ClusteredIT {
+
+   @Substitute
+   public static JavaArchive[] artifacts() {
+      return new JavaArchive[0];
```

Comment:
> Can we add some documentation here as to why this works? Where are all these classes and jars added?

They aren't :slightly_smiling_face:. With this approach we have to avoid tests that rely on server-side dependencies. It might be possible to get Shrinkwrap to work with native as expected, but I didn't have much time to investigate.

I've added a comment to the class.